### PR TITLE
Identify profiles by ID 

### PR DESCRIPTION
**Pull request by MarcMichalsky**
Storing profiles in a separate db table instead of native CiviCRM settings was implemented in fd47b91b65cd1e6acf3387b642678abfbcd1e079. 

Nevertheless, the name of the profile is still used to identify it. This leads to problems like #70.

I have now implemented the identification of profiles by their database ID.

**This PR depends on PR #72 because it uses the custom `ProfileException` class!**

### CODE REVIEW

File: CRM/Twingle/Form/Profile.php
Comment (jensschuppe) [inline CRM/Twingle/Form/Profile.php:160]: This should check vor a valid profile instead, as `$profile_id` is basically just an integer retrieved from the request.

File: CRM/Twingle/Form/Profile.php
Comment (jensschuppe) [inline CRM/Twingle/Form/Profile.php:180]: This should be validated, as we can't be sure the passed `source_profile` value ist actually a valid profile.

File: CRM/Twingle/Profile.php
Comment (jensschuppe) [inline CRM/Twingle/Profile.php:83]: I'd vote for the `selector` property being kept when copying profiles, as it's technically just another piece of `$data`.

File: CRM/Twingle/Profile.php
Comment (jensschuppe) [inline CRM/Twingle/Profile.php:300]: Should this get a specific error code (`..._DELETE_PROFILE` instead of `..._RESET_PROFILE`)?

**Review by jensschuppe — CHANGES_REQUESTED** [review]

File: CRM/Twingle/Profile.php
Comment (MarcMichalsky) [reply CRM/Twingle/Profile.php:83]: But a selector should only be used in one profile, right?

File: CRM/Twingle/Profile.php
Comment (MarcMichalsky) [reply CRM/Twingle/Profile.php]: Yep, sorry.

File: CRM/Twingle/Profile.php
Comment (MarcMichalsky) [reply CRM/Twingle/Profile.php]: Oh, the `ERROR_CODE_COULD_NOT_DELETE_PROFILE` already exists. Nice catch!

File: CRM/Twingle/Form/Profile.php
Comment (MarcMichalsky) [reply CRM/Twingle/Form/Profile.php:160]: It does not come only from the query parameters. At this stage it has already been validated in the `CRM_Twingle_Profile::getProfile($this->profile_id);` method (line 152) which returns NULL if the profile does not exist.

Comment (MarcMichalsky) [reply CRM/Twingle/Form/Profile.php:160]: Sorry, my fault. 😑
I will fix that.

Comment (MarcMichalsky) [reply CRM/Twingle/Form/Profile.php]: > PS. cross-posted with your edited answer.
Yes, sorry for the confusion. I only realized my mistake after sending the comment.

**Comment by jensschuppe** [comment]
@MarcMichalsky I rebased onto current `master` which has the required changes from #72 - could you have a look whether you can spot anything odd? There were some merge conflicts I resolved but I would like to get another pair of eyes on it before merging …

**Comment by MarcMichalsky** [comment]
I'll have a look at it next week. 🏝️

**Pull request merged**
